### PR TITLE
Refactor intent layers

### PR DESCRIPTION
**Pull request by janaab11**
- Repeating code blocks are pulled out as methods and reusable variables (larger scope)
- Added comments for clarity

### CODE REVIEW

File: eevee/metrics/classification.py [hunk 1]
```diff
@@ -3,6 +3,8 @@
 import pandas as pd
 from sklearn.metrics import classification_report, precision_recall_fscore_support
 
+TRUE = "intent_x"
```
Comment (lepisma) [inline eevee/metrics/classification.py:6]: Hard to find this variable at top level and figure out what it is. Maybe omething like 'TRUE_LABEL' would be better

File: eevee/metrics/classification.py [hunk 2]
```diff
@@ -98,55 +103,103 @@ def intent_report(
...
-            group_classification_report = classification_report(df["{}-alias".format("intent_x")],
-                                                                df["{}-alias".format("intent_y")],
-                                                                labels=[sub_layer], output_dict=return_output_as_dict, zero_division=0)
-            group_classification_report_df = pd.DataFrame(group_classification_report).transpose()
-            group_classification_report_df["support"] = group_classification_report_df["support"].astype('int32')
-            grouped_classification_reports["layer-{}".format(sub_layer)] = group_classification_report_df
+        for sub_layer in intent_layers.get(TRUE):
+
+            col = PREDICTED
+            df[ALIAS_SUFFIX.format(col)] = df[col].apply(lambda intent: {PREDICTED_LAYER: sub_layer}.get(intent, intent))
```
Comment (greed2411) [inline eevee/metrics/classification.py:188]: i find these two lines repeating a lot and confusing. this is the only thing holding me back.

`PREDICTED_LAYER` is a `str` (constant throughout), `sub_layer` is also a `str`. then why the dict of 
`{PREDICTED_LAYER: sub_layer}.get(intent, intent)` ?

is this what you are trying to do?
```
lambda intent: sub_layer if PREDICTED_LAYER == intent else intent
```

Comment (janaab11) [reply eevee/metrics/classification.py:188]: **Short answer**, yes that what i am doing

**Long answer**. _why_ i am using a dictionary here, is to maintain regularity with what i do above this piece of code (this holds for wherever i used this piece of code)

``` python
df[ALIAS_SUFFIX.format(col)] = df[col].apply(lambda intent: intents_dict.get(intent, intent))
```

This line, and the one you referenced go together. Below, i'll try to explain why - for ease of reference, let's call the above `line 1` and the one you referenced, `line 2`.

The goal of this PR is to calculate some numbers for sub-layers of a specific layer (for eg. acoustic and lexical are sub layers of the OOS layer). The assumption here, and this is what makes this PR necessary, is that only the `TRUE` tags contain the sub-layer intents, while the `PREDICTED` tags only contain the layer intent(s). 
Here, line 1 acts on the `TRUE` tags of the data, and aliases intents to the respective sub-layers. Now for each sub-layer, we want to calculate metrics. So `line 2` aliases the layer intent(s) in `PREDICTED` tags to the selected sub-layer. Keeping it a dictionary in `line 2` is to, both:

1. maintain regularity across pieces of code that do the same thing (in this case, aliasing)
2. allow flexibility for multiple intents in the layer, in the future

**Comment by greed2411** [comment]
have made changes to the code as discussed.

still have two more things:
* tests for layer  intents
* docs for layer intents

once i write them to this branch, i'll merge it to dev. @janaab11

**Comment by greed2411** [comment]
added docs & tests: requesting @janaab11 to verify if the docs makes sense. if not please give suggestions to fix them.

**Pull request merged**